awx/api/serializers.py

```diff
@@ -2603,13 +2605,31 @@ def get_summary_fields(self, obj):
 
         return summary_dict
 
+    def validate(self, attrs):
+        if self.instance and self.instance.managed_by_tower:
+            raise PermissionDenied(
+                detail=_("Modifications not allowed for managed credentials")
+            )
+        return super(CredentialSerializer, self).validate(attrs)
+
     def get_validation_exclusions(self, obj=None):
         ret = super(CredentialSerializer, self).get_validation_exclusions(obj)
         for field in ('credential_type', 'inputs'):
             if field in ret:
                 ret.remove(field)
         return ret
 
+    def validate_organization(self, org):
+        if (
+            self.instance and
+            self.instance.credential_type.kind == 'galaxy' and
+            org is None
+        ):
+            raise serializers.ValidationError(_(
+                "Galaxy credentials must be owned by an Organization."
```

Comment:
What if the org is deleted?

Reply:
I don't want to muck with the credential organization set null behavior

https://github.com/ansible/awx/blob/f4af74dabebc4d716ffe470b834c23e293f5c94f/awx/main/models/credential/__init__.py#L103-L109

So it would leave it null. I'm not too terribly bothered by that. As long as an org admin can still _remove_ it, I'm okay with it.

(note there are 2 orgs in play here, the one who owns the credential and the one using the credential, I am talking about the ownership org being deleted)